utilities/internal_time_types_test.go

```diff
@@ -0,0 +1,130 @@
+package utilities
+
+import (
+	"encoding/json"
+	"testing"
+	"time"
+)
+
+func TestUnixDateTimeString(t *testing.T) {
+	newTime := time.Unix(1484719381, 0)
+	unixDateTime := NewUnixDateTime(newTime)
+	AssertEquals(t, "1484719381", unixDateTime.String())
+}
+
+func TestUnixDateTime(t *testing.T) {
+	newTime := time.Unix(1484719381, 0)
+	unixDateTime := NewUnixDateTime(newTime)
+
+	bytes, err := json.Marshal(unixDateTime)
+	AssertNoError(t, err)
+	var newUnixDateTime UnixDateTime
+	err = json.Unmarshal(bytes, &newUnixDateTime)
+	AssertNoError(t, err)
+	AssertEquals(t, newTime, newUnixDateTime.Value())
+}
+
+func TestUnixDateTimeError(t *testing.T) {
+	var newUnixDateTime UnixDateTime
+	json.Unmarshal([]byte(`"Sun, 06 Nov 1994 08:49:37 GMT"`), &newUnixDateTime)
+}
+
+func TestDefaultTimeString(t *testing.T) {
+	newTime, err := time.Parse(DEFAULT_DATE, "1994-02-13")
+	AssertNoError(t, err)
+	defaultTime := NewDefaultTime(newTime)
+	AssertEquals(t, "1994-02-13", defaultTime.String())
+}
+
+func TestDefaultTime(t *testing.T) {
+	newTime, err := time.Parse(DEFAULT_DATE, "1994-02-13")
+	AssertNoError(t, err)
+	defaultTime := NewDefaultTime(newTime)
+
+	bytes, err := json.Marshal(defaultTime)
+	AssertNoError(t, err)
+	var newDefaultTime DefaultTime
+	err = json.Unmarshal(bytes, &newDefaultTime)
+	AssertNoError(t, err)
+	AssertEquals(t, newTime, newDefaultTime.Value())
+}
+
+func TestDefaultTimeError1(t *testing.T) {
+	var newDefaultTime DefaultTime
+	json.Unmarshal([]byte(`1484719381`), &newDefaultTime)
```

Comment:
No assertions are being made in the Error Tests. Please use AssertError(t, err) function for all

Reply:
added changes for error assertion.